bleak/backends/bumble/scanner.py

```diff
@@ -0,0 +1,124 @@
+from bumble.device import Advertisement, Device
+from bumble.hci import Address
+from bumble.host import Host
+
+from bleak.backends.bumble import (
+    BumbleTransport,
+    get_default_transport,
+    get_link,
+    start_transport,
+)
+from bleak.backends.scanner import (
+    AdvertisementData,
+    AdvertisementDataCallback,
+    BaseBleakScanner,
+)
+
+logger = logging.getLogger(__name__)
+
+SERVICE_UUID_TYPES = [
+    AdvertisingData.COMPLETE_LIST_OF_128_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.COMPLETE_LIST_OF_16_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.COMPLETE_LIST_OF_32_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.INCOMPLETE_LIST_OF_16_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.INCOMPLETE_LIST_OF_32_BIT_SERVICE_CLASS_UUIDS,
+    AdvertisingData.INCOMPLETE_LIST_OF_128_BIT_SERVICE_CLASS_UUIDS,
+]
+
+SCANNER_MAC_ADDR = "F0:F1:F2:F3:F4:F5"
+
+
+class BumbleScanner(BaseBleakScanner, Device.Listener):
+    """
+    Interface for Bleak Bluetooth LE Scanners
+
+    Args:
+        detection_callback:
+            Optional function that will be called each time a device is
+            discovered or advertising data has changed.
+        service_uuids:
+            Optional list of service UUIDs to filter on. Only advertisements
+            containing this advertising data will be received.
+    Keyword Args:
+        adapter (BumbleTransport): Bumble transport adapter to use.
+    """
+
+    def __init__(
+        self,
+        detection_callback: Optional[AdvertisementDataCallback],
+        service_uuids: Optional[List[str]],
+        scanning_mode: Literal["active", "passive"],
+        **kwargs,
+    ):
+        super(BumbleScanner, self).__init__(detection_callback, service_uuids)
+        self.device = Device("scanner_dev", address=Address(SCANNER_MAC_ADDR))
+
+        self.device.host = Host()
+        self.device.host.controller = Controller("Scanner", link=get_link())
+        self.device.listener = self
+        self._scan_active = scanning_mode == "active"
+        self._adapter: BumbleTransport = kwargs.get("adapter", get_default_transport())
+
+    def on_advertisement(self, advertisement: Advertisement):
+        service_uuids: List[str] = []
+        service_data: Dict[str, AdvertisingDataObject] = {}
+        local_name = advertisement.data.get(AdvertisingData.COMPLETE_LOCAL_NAME)
+        if not local_name:
+            local_name = advertisement.data.get(AdvertisingData.SHORTENED_LOCAL_NAME)
+        manuf_data = advertisement.data.get(AdvertisingData.MANUFACTURER_SPECIFIC_DATA)
+        for uuid_type in SERVICE_UUID_TYPES:
+            adv_uuids = advertisement.data.get(uuid_type)
+            if adv_uuids is None:
+                continue
+            if not isinstance(adv_uuids, list):
+                continue
+            for uuid in adv_uuids:
+                if uuid not in service_uuids:
+                    service_uuids.append(str(uuid))
```

Comment:
The type of `adv_uuids` is a List of `Bumble.Core.UUID`.  Which is not a normalized uuid but a custom class from bumble.

I have to convert it to string to get the string representation, otherwise the implementation specific class for a UUID from Bumble is used.